src/components/FieldSet.tsx

```diff
@@ -0,0 +1,25 @@
+/* @jsx jsx */
+import { jsx } from '@emotion/core';
+import * as React from 'react';
+
+import { Box, IBox } from '@stoplight/ui-kit';
+
+import { FormtronComponentVariant } from '../types';
+import { useBorder } from './hooks';
+
+interface IFieldSet extends IBox<HTMLFieldSetElement> {
+  legend: HTMLElement | string;
+  variant?: FormtronComponentVariant;
+}
+
+export const FieldSet: React.FunctionComponent<IFieldSet> = ({ legend, variant, children }) => {
+  const borderColor = useBorder(variant);
+  return (
+    <Box as="fieldset" border={`1px solid ${borderColor || 'currentColor'}`}>
```

Comment:
```suggestion
    <Box as="fieldset" border={`1px solid ${borderColor}`}>
```

Reply:
Can't. `1px solid undefined` syndrome.

```
  input: {
    fg: string;
    bg: string;
    border?: string;
    invalidFg?: string;
    invalidBg?: string;
    invalidBorder?: string;
  };
```

We could move `|| 'currentColor'` into `useBorder` but that logic might not always make sense.

Reply:
See comment above. You won't get undefined.

Reply:
which comment above? You totally get undefined, because `input.border` is not defined. That's the reason I added `|| 'currentColor'` in the first place.

Reply:
![image](https://user-images.githubusercontent.com/9273484/53382335-904f9600-39a6-11e9-9e99-bff33a54a5dd.png)

^, btw would make sense to add a default value for `useSth` helper? Same as lodash.get does.

Reply:
Ah. You overwhelmed the system I guess. Didn't realize it would auto-hide important comments... thought it was just resolved ones that got hid.

![image](https://user-images.githubusercontent.com/587740/53418656-45358300-39a6-11e9-9360-c01751fed4e7.png)